dll/win32/msafd/misc/sndrcv.c

```diff
@@ -203,6 +204,11 @@ WSPRecv(SOCKET Handle,
     if( !NT_SUCCESS(Status) )
         return -1;
 
+    /* HACK: change the Socket to Non Blocking. See CORE-12828. */
+    BlockMode = TRUE;
+    SetSocketInformation(Socket, AFD_INFO_BLOCKING_MODE, &BlockMode, NULL, NULL, NULL, NULL);
+    Socket->SharedData->NonBlocking = TRUE;
```

Comment:
I seriously doubt Windows' `WSPRecv` really tampers with the `NonBlocking` member state of the socket (do you have any user testcase that exactly shows Windows really does that?). Taking a look at the function on a first glance it seems it merely receives data from the socket so it feels kind of weird that you're toying with the socket in that way.

Reply:
OK, I just did some investigation. I guess that in Windows, the `NonBlocking` member is always set to `TRUE` in `WSPSocket` instead, unlike our implementation. ROS shows me the following offsets:
`fixme:(dll/win32/msafd/misc/dllmain.c:76) Offset of NonBlocking is 35, offset of SharedData is 4`
when I'm specifying the following:
`FIXME("Offset of NonBlocking is %x, offset of SharedData is %x\n", offsetof(SOCK_SHARED_INFO, NonBlocking), offsetof(SOCKET_INFORMATION, SharedData));`
How can I calculate the offset properly? :thinking: I guess I should just append the two offsets. If to append the 1st offset to the 2nd, then it seems to be 39.
The most related to this value which I found in unassemble of `WSPSocket` function, is 40. There is specified an `ebx` register with appending this offset and the `BOOLEAN` value to `TRUE` (1):
![nonblocking_investigations](https://user-images.githubusercontent.com/26385117/143085056-ce3a07f0-9077-4247-819a-9a69a77ff20f.png)
The total offset is higher on 1 probably because there is some additional member in any of these structures, about which we don't know. But it's just my guessing.
**This judgement is valid only in case I calculated the offset correctly.** If no, then how can I calculate it correctly instead? :thinking: